app/controllers/messages_controller.rb

```diff
@@ -127,6 +134,23 @@ def mark
     render :action => "no_such_message", :status => :not_found
   end
 
+  # Moves message into Inbox by unsetting the muted-flag
+  def unmute
+    message = Message.where(:recipient => current_user).find(params[:message_id])
```

Comment:
We prefer `find_by` instead of `where().first`, but `Model.find(id)` is still preferred to `Model.find_by(:id => id)`.

I think some kind of scoping is necessary to avoid unmuting other people's messages, although I'd normally try to find a `current_user.something.find()` to scope them rather than building a where query.

Reply:
> This never actually changes the state of the message, so nothing happens when you click the Unmute button for a message in the user interface.

this is just the query for the `message`-record. 2 lines below `message.unmute` is called.

Reply:
> I think some kind of scoping is necessary to avoid unmuting other people's messages

The query is already scoped to the `current_user`

```rb
Message.where(:recipient => current_user).find(params[:message_id])
```

But yes, I prefer a scope on the `current_user`, too. I'll change this too:

```rb
current_user.muted_messages.find(params[:message_id])
```

Reply:
> this is just the query for the `message`-record. 2 lines below `message.unmute` is called.

Oh I missed that... In that case the question becomes why wasn't it working for me because the reason I stared looking closely at that code was that the unmute didn't seem to work.

Reply:
> In that case the question becomes why wasn't it working for me

not sure what went wrong there :)